sync-core/src/main/java/com/cloudant/sync/query/IndexCreator.java

```diff
@@ -252,6 +268,32 @@ protected static boolean validFieldName(String fieldName) {
         return result;
     }
 
+    /**
+     * Based on the proposed index and the list of existing indexes, this method checks
+     * whether another index can be created.  Currently the limit for TEXT indexes is 1.
+     * JSON indexes are unlimited.
+     *
+     * @param index the proposed index
+     * @param existingIndexes the list of already existing indexes
+     * @return whether the index limit has been reached
+     */
+    @SuppressWarnings("unchecked")
+    protected static boolean indexLimitReached(Index index, Map<String, Object> existingIndexes) {
+        if (index.indexType.equalsIgnoreCase(Index.TEXT_TYPE)) {
+            for (String name : existingIndexes.keySet()) {
+                Map<String, Object> existingIndex = (Map<String, Object>) existingIndexes.get(name);
+                String type = (String) existingIndex.get("type");
+                if (type.equalsIgnoreCase("text") && !name.equalsIgnoreCase(index.indexName)) {
```

Comment:
All we're really interested here is to check whether another *text* index already exists.  A later check for name and field equality is performed and in the scenario where the name is the same but the fields are different, this check fails, which drops us into the table creation logic which will in turn also fail since a table does already exist albeit with different data structures.  This logic is untouched since the port of your original implementation.